architecture/2024-08-controller-component-threat-model.md

```diff
@@ -0,0 +1,186 @@
+# Title
+
+- **Author**: ytimocin
+
+## Overview
+
+This document aims to provide a threat model for the Radius Controller component. In this threat model, you will find potential security threats to the Controller component, an important part of Radius, and possible mitigations to the risks they pose. The document includes an analysis of the system, its assets, security objectives, identified threats, and recommended security measures to protect the system.
+
+This document will not include designs for the proposed measures to each threat.
+
+## System Description
+
+The Controller component is where we keep our Kubernetes controllers, a validating webhook, and a few other important parts. We will dive into more details on the controller below.
+
+Note: If you would like to learn more about Kubernetes controllers, you can visit [this link](https://kubernetes.io/docs/concepts/architecture/controller/).
+
+### Architecture
+
+The Controller component consists of several key parts:
+
+- **Recipe and Deployment Controllers**: Custom Kubernetes controllers registered with the manager. Each controller is responsible for watching specific Kubernetes resources (Recipe and Radius Deployment) and reconciling their state.
+- **Recipe Validating Webhook**: [Used if TLS certificates are provided] This webhook is triggered by the Kubernetes API in case of a change in a Recipe resource. Kubernetes API reaches out to this webhook in case of a creation, an update, or a deletion of a Recipe. Webhook validates the action, if it is a valid action, and responds to the Kubernetes API. For more information about webhooks, refer to the [official documentation](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/).
+- **Health Checks**: Health checks are implemented to monitor the status and performance of the controllers. They ensure that the controllers are functioning correctly and can take corrective actions if any issues are detected.
+
+### Clients
+
+In this section we are going to be talking about different clients of the Controller component. Clients are systems that reach out to the Controller component to trigger an action. Here are the clients of the Controller component:
+
+1. **Kubernetes API Server**: The primary client that interacts with the controller. It sends events related to resource changes (e.g., creation, update, or deletion of a Recipe and/or Deployment) to the controller manager. The controller watches for these events and reconciles the state of the resources accordingly.
+1. **Validating Webhook Clients**: If webhooks are enabled and registered, clients that interact with the Kubernetes API server (e.g., `kubectl`, other controllers) can trigger webhook calls.
```

Comment:
> If webhooks are enabled and registered

Is this something that users get to configure?

> clients that interact with the Kubernetes API server (e.g., `kubectl`, other controllers) can trigger webhook calls

Do these "other clients" interact with our webhook directly?

Reply:
I don't think any client other than Kubernetes API Server can talk to the Validating Webhook. Not sure if we need to specify this in the configuration of the Webhook: https://github.com/radius-project/radius/blob/main/deploy/Chart/templates/controller/validating-webhook-configuration.yaml#L28.

Based on what I have been reading for the last 20 minutes, Kubernetes API Server is the only client that can access a given Webhook. Correct me if I am wrong, @rynowak.